src/Core/Cache/ApcuCacheFactory.php

```diff
@@ -27,9 +27,13 @@ public function __construct($version = null)
      */
     public function create($service, array $params = array())
     {
+        $namespace = isset($params['namespace']) ? $params['namespace'] : '';
+        $defaultLifetime = isset($params['defaultLifetime']) ? $params['defaultLifetime'] : 0;
+        $directory = isset($params['directory']) ? $params['directory'] : null;
+        $apcuNamespace = $namespace . '_' . substr(md5($directory), 0, 7);
```

Comment:
7 characters on an md5 is sufficiently rare that no two instances of silverstripe on a site will collide. there won't be any bad characters, but I can increase it if needed.